bot.py

```diff
@@ -29,492 +20,75 @@
 
 """
 
-
-# Slack imports
 from slackclient import SlackClient
-# Google imports
-from oauth2client.service_account import ServiceAccountCredentials
-from httplib2 import Http
-from apiclient.discovery import build
-# Other imports
+import threading, websocket, json, re, time, codecs, random, logtofile
 import scripts
 from scripts import commands
-import os, io, sys, time, codecs, websocket, json, logging, random, logtofile  # noqa: 401
-import re
-
-# Version Number: release.version_num.revision_num
-VERSION = "1.1.9"
-
-# Config Variables
-BOT_NAME = ""
-BOT_ICON_URL = ""
-SLACK = False
-GOOGLECAL = False
-LOGGER = False
-GOOGLECALSECRET = ""
-NEWUSERGREETING = False
-GREETING = ""
-EMOJI_LIST = ["party-parrot", "venezuela-parrot", "star2", "fiesta-parrot", "wasfi_dust", "dab"]  # noqa: 501
-USER_LIST = []
-ADMIN = []
-ADMIN_COMMANDS = ["log", "calendar", "admin"]
-TTPB = ""
-GENERAL = ""
-
-# Global variabls
-global LOG
-LOG = False
-global LOGC
-LOGC = []
-global pbCooldown
-pbCooldown = 100
-polling_list = dict()
-
-
-# function that is called whenever there is an event, including status changes, join messages, typing status, emoji reactions, everything  # noqa: 501
-def on_message(ws, message):
-
-    s = message.decode('utf-8')
-    response = json.loads(unicode(s))
-    print "PantherBot:LOG:message:" + response["type"]
-
-    # Pugbomb cooldown incrementation
-    global pbCooldown
-    pbCooldown += 1
-    global LOG
-    global LOGC
-
-    # Checks if the event type returned by Slack is a message
-    if "message" == response["type"]:
-        if "subtype" in response:
-            if response["subtype"] == "bot_message":
-                #polling logic
-                if "POLL_BEGIN" in response["text"]:
-                    polling_list[response["channel"]][0] = response["ts"]
-                    temp_options = polling_list[response["channel"]][1]
-                    for key in temp_options:
-                        rreaction(response, temp_options[key].strip(":"))
-                return
-        # Checks LOG and LOGC values, then
-        # If LOG has been set to true it will save all spoken messages.
-        check_log()
-        if LOG and response["channel"] in LOGC:
-            logtofile.log(sc, response)
 
-        # Announcement reactions
-        react_announcement(response)
-        # if riyansDenial(response):
-        #     return
-        if other_message(response):
-            return
-        if command_message(response):
-            return
-        if admin_message(response):
-            return
-
-    elif "team_join" == response["type"]:
-        if NEWUSERGREETING:
-            new_user_message(response)
-        USER_LIST = sc.api_call(  # noqa: 841
-            "users.list",
+class Bot(object):
+    ADMIN = ["U25PPE8HH", "U118FGXFA", "U0LAMSXUM", "U3EAHHF40"]
```

Comment:
Much agreed, we have talked about environment variable options, tell me the format we want and I'll update as needed. I'm thinking `PB_ADMIN` as csv?